File: src/main/java/com/fastcampus/innercircle/survey_api/common/Response.java
package com.fastcampus.innercircle.survey_api.common;

public class Response<T> {

    private final T data;

    public T getData() {
        return data;
    }

    public Response(T data) {
        this.data = data;
    }

    public static <T> Response<T> of(T data) {
        return new Response<>(data);
    }
}

File: survey-domain/src/main/kotlin/com/innercircle/surveydomain/model/Response.kt
@file:Suppress("ktlint:standard:no-wildcard-imports")

package com.innercircle.surveydomain.model

import jakarta.persistence.*

@Entity
@Table(name = "responses")
data class Response(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int = 0,
    @ManyToOne @JoinColumn(name = "question_id", nullable = false)
    val question: Question,
    @ManyToOne @JoinColumn(name = "submission_id", nullable = false)
    val submission: Submission,
    @ManyToOne
    @JoinColumn(name = "question_option_id")
    val questionOption: QuestionOption?,
    @Column(columnDefinition = "TEXT")
    val responseData: String? = null,
)
